src/cli/cli.cpp

```diff
@@ -0,0 +1,123 @@
+#include "cli.hpp"
+#include "states.hpp"
+#include "Particle.h"
+#include "conio.hpp"
+#include "menuItems/systemCommands.hpp"
+
+#include <fstream>
+
+const CLI_menu_t CLI_menu[] =
+{
+    {"menu", "display Menu", &CMD_displayMenu},
+    {"disconnect", "disconnect particle", &CLI_disconnect},
+    {"connect", "connect particle", &CLI_connect},
+    {"showerrors", "show flog errors", &CLI_displayFLOG},
+    {NULL,NULL,NULL}
+};
+
+static STATES_e CLI_nextState;
+
+void CLI::init(void) 
+{
+    CLI_nextState = STATE_CLI;
+
+    // While there is an avaliable character typed, get it
+    while(kbhit())
+    {
+        getch();
+    }
+}
+
+STATES_e CLI::run(void)
+{
+    CLI_menu_t *cmd;
+    char* userInput = new char[100];
```

Comment:
We should avoid dynamic memory allocation.  It would be better for this to be a module scope or object scope statically allocated buffer that we `memset` to 0 before using.

Reply:
Also good idea, also fixed